### DISCUSSION TITLE
root.json expired

### DISCUSSION THREAD

**Discussion by abenzick** [dennisvang/tufup#119] [Q&A]
I let the root.json file expire and now my app is erroring out on load. Will I have to publish a new build or can I update the root.json on the back-end to keep things going?

**Comment by dennisvang** [comment]
@abenzick It should be sufficient to re-sign the `root.json` file.

(instructions will follow)

**Comment by dennisvang** [comment]
On the command line (also see output of `tufup sign --help`):

```
tufup sign -e <number of days until new expiration> root <path to root key(s)>
```

This will update the expiration date and re-sign the `root.json` file. You'll end up with an updated `root.json` and a new version, e.g. `2.root.json` (which is a copy of the updated `root.json`). 

Note that ***all*** versions of `root.json` should be made available on your update-server. This is [explained in the TUF spec][1].

[1]: https://theupdateframework.github.io/specification/latest/#update-root

**Comment by abenzick** [comment]
> On the command line (also see output of `tufup sign --help`):
> 
> ```
> tufup sign -e <number of days until new expiration> root <path to root key(s)>
> ```
> 
> This will update the expiration date and re-sign the `root.json` file. You'll end up with an updated `root.json` and a new version, e.g. `2.root.json` (which is a copy of the updated `root.json`).
> 
> Note that [_all_ versions](https://theupdateframework.github.io/specification/latest/#update-root) of `root.json` should be made available on your update-server.

Back in the green! Simple and painless. THANK YOU FOR THE PROMPT RESPONSE. You saved me a lot of headaches today. Do you have a donation page?

**Comment by dennisvang** [comment]
>Back in the green! ...

That's good to hear. Thanks for the feedback!

>Do you have a donation page?

No, I don't, but thanks for the kind gesture. I'm happy to help. :)
